### ISSUE TITLE
Discriminator training in wgan gp mode

### ISSUE THREAD

**Issue by CBD88** [junyanz/pytorch-CycleGAN-and-pix2pix#1050]
It is advisable to train discriminator with the same image for more iteration as mentioned in [https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/357](url) (here it is generator but i want discriminator to train more)  like 

self.set_requires_grad(self.netD, True)
    self.optimizer_D.zero_grad()
    self.backward_D()
    self.optimizer_D.step()
    for i in range(20):
          # update G
          self.set_requires_grad(self.netD, False)
          self.optimizer_G.zero_grad()
          self.backward_G()
          self.optimizer_G.step()

OR 

Will it be good if i take different image?

**Comment by junyanz** [comment]
It might be better to take different images. But the difference should be quite small. You can also just increase the learning rate for D.

**Comment by CBD88** [comment]
Okay. Thanks.
